### PR TITLE
[Feature] Get the correct permissions after switching team

### LINKED ISSUE #1844: [Feature] Get the correct permissions after switching team
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Currently, the permission isn't changed when switch to other team.

For example, user_a is a developer of team_1, and he is a team_admin of team_2.

When choose team from `team_1` to `team_2`, the front-end should get the correct  permissions from backend.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by 1996fanrui**
## What changes were proposed in this pull request

Issue Number: close #1844 

Get the correct permissions after switching team

## Brief change log

Get the correct permissions after switching team

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): no

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/MenuController.java [hunk 1]
```diff
@@ -53,8 +53,8 @@ public class MenuController {
     private CommonService commonService;
 
     @PostMapping("router")
-    public RestResponse getUserRouters() {
-        ArrayList<VueRouter<Menu>> routers = this.menuService.getUserRouters(commonService.getCurrentUser());
+    public RestResponse getUserRouters(@NotBlank(message = "{required}") Long teamId) {
+        ArrayList<VueRouter<Menu>> routers = this.menuService.getUserRouters(commonService.getCurrentUser().getUserId(), teamId);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/MenuController.java]: The code can be further improve:
`ArrayList<VueRouter<Menu>> routers = this.menuService.getUserRouters(commonService.getUserId(), teamId);`

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/PassportController.java [hunk 2]
```diff
@@ -124,8 +124,6 @@ private Map<String, Object> generateUserInfo(JWTToken token, User user) {
         Set<String> roles = this.roleService.getUserRoleName(username);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/PassportController.java]: username is not recommended, userId is recommended

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/UserService.java [hunk 3]
```diff
@@ -103,7 +105,16 @@ public interface UserService extends IService<User> {
      */
     void resetPassword(String[] usernames) throws Exception;
 
-    Set<String> getPermissions(String username);
+    Set<String> getAllTeamPermissions(String username);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/UserService.java:110]: I don't think it's necessary to add this method

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -159,9 +160,14 @@ public void resetPassword(String[] usernames) {
      * @return permissions
      */
     @Override
-    public Set<String> getPermissions(String username) {
-        List<Menu> permissionList = this.menuService.findUserPermissions(username);
-        return permissionList.stream().map(Menu::getPerms).collect(Collectors.toSet());
+    public Set<String> getAllTeamPermissions(String username) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java:170]: `getPermissions` method can replaced `getAllTeamPermissions` method, we don't need this method, What do you think?

**Comment by 1996fanrui** [comment]
Hi @wolfboys , I updated this PR base on #1868 , please help review in your free time, thanks~

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**